sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -0,0 +1,290 @@
+
+        Notes
+        -----
+        This method should not include following parameters:
+
+        - future exogenous columns (``futr_exog_list``) - used from ``__init__``
+        - historical exogenous columns (``hist_exog_list``) - not supported
+        - statis exogenous columns (``stat_exog_list``) - not supported
+        - custom model name (``alias``) - used from ``algorithm_name``
+        """
+
+    def _instantiate_model(self: "_NeuralForecastAdapter", fh: ForecastingHorizon):
+        """Instantiate the model."""
+        exogenous_parameters = (
+            {"futr_exog_list": self.futr_exog_list} if self.needs_X else {}
+        )
+
+        algorithm_instance = self.algorithm_class(
+            fh,
+            alias=self.algorithm_name,
+            **self.algorithm_parameters,
+            **exogenous_parameters,
+        )
+
+        from neuralforecast import NeuralForecast
+
+        model = NeuralForecast(
+            [algorithm_instance], self.freq, local_scaler_type=self.local_scaler_type
+        )
+
+        return model
+
+    def _fit(
+        self: "_NeuralForecastAdapter",
+        y: pandas.Series,
+        X: typing.Optional[pandas.DataFrame],
+        fh: ForecastingHorizon,
+    ) -> "_NeuralForecastAdapter":
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : sktime time series object
+            guaranteed to have a single column/variable
+        fh : guaranteed to be ForecastingHorizon
+            The forecasting horizon with the steps ahead to to predict.
+        X : sktime time series object, optional (default=None)
+            guaranteed to have at least one column/variable
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : _NeuralForecastAdapter
+            reference to self
+        """
+        if not fh.is_all_out_of_sample(cutoff=self.cutoff):
+            raise NotImplementedError("in-sample prediction is currently not supported")
+
+        train_indices = y.index
+        if isinstance(train_indices, pandas.PeriodIndex):
+            train_indices = train_indices.to_timestamp(freq=self.freq)
+
+        train_data = {
+            self.id_col: 1,
+            self.time_col: train_indices.to_numpy(),
+            self.target_col: y.to_numpy(),
+        }
+
+        if self.futr_exog_list and X is None:
+            raise ValueError("Missing exogeneous data, 'futr_exog_list' is non-empty.")
+
+        if self.futr_exog_list:
+            for column in self.futr_exog_list:
+                train_data[column] = X[column].to_numpy()
+
+        train_dataset = pandas.DataFrame(data=train_data)
+
+        maximum_forecast_horizon = fh.to_relative(self.cutoff)[-1]
+        self._forecaster = self._instantiate_model(maximum_forecast_horizon)
```

Comment:
Just a small question, should this be `self.forecaster_`, given that the docstrings for fit and predict say *Fitted model attributes ending in "_"*?

Reply:
As of now, all other adapters (statsmodels/pmdarima/etc.) use `self._forecaster`. I've not changed here or in the `statsforecast` adapter to remain consistent.

I think the reason for not doing in adapters is that having underscore at the end updates the results of `get_fitted_params()`, and fitted model itself is not a parameter. This is my guess, may be @fkiraly / @benHeid can confirm if this was the reason for earlier adapters or something else.

But having the model itself, or if it has some other properties like AIC/BIC/etc., may be useful. I'll wait for @fkiraly / @benHeid to comment for fitted model, but if you can suggest some other quick model properties that may be useful for users, I can override `_get_fitted_params` in the meanwhile.

Reply:
really depends on what you would like to expose to the user. It is a valid choice to not expose the adapted class, as long as most fitted paraemters are forwarded, but I think it would also be valid to do so.

As @yarnabrina says, in the current codebase, components (parameters) usually have their fitted clones exposed as a fitted parameters, but adaptees not, although I don't think there's a strict rule on this.